tests/serverpod_test_server/test_integration/method_streaming_cancel_test.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:test/test.dart';
+
+import 'test_tools/serverpod_test_tools.dart';
+
+void main() {
+  withServerpod(
+    'Given call to MethodStreamingEndpoint',
+    (endpoints, session) {
+      test(
+          'when calling an endpoint returning a non-broadcast stream and cancelling '
+          'then will cancel', () async {
+        var stream = endpoints.methodStreaming.intStreamFromValue(session, 1);
+        var subscription = stream.listen((event) {});
+
+        await expectLater(subscription.cancel(), completes);
+      });
+
+      test(
+          'when calling an endpoint returning a broadcast stream and cancelling '
+          'then it should cancel and trigger the onCancel hook on the stream controller',
+          () async {
+        var wasStreamCancelled =
+            endpoints.methodStreaming.wasBroadcastStreamCanceled(session);
+
+        var stream = endpoints.methodStreaming.getBroadcastStream(session);
+        // Wait for the stream to be created, otherwise cancel is called before creation
+        await flushMicrotasks();
+
+        var subscription = stream.listen((event) {});
+        await subscription.cancel();
+
+        await expectLater(
+          wasStreamCancelled,
+          completion(true),
+        );
+      });
+
+      test(
+          'when calling an endpoint returning a broadcast stream and cancelling '
+          'then it should close the session and call its will close listener',
+          () async {
+        var wasSessionWillCloseListenerCalled = endpoints.methodStreaming
+            .wasSessionWillCloseListenerCalled(session);
+
+        var stream = endpoints.methodStreaming.getBroadcastStream(session);
+        // Wait for the stream to be created, otherwise cancel is called before creation
+        await flushMicrotasks();
```

Comment:
Comment: Microtasks are separate from events in the flutter async loop. I think the intention is to flush all events in the event loop right? Might need to clarify the naming of this method in that case: https://web.archive.org/web/20170704074724/https://webdev.dartlang.org/articles/performance/event-loop

Reply:
Good catch, will rename in this pr: https://github.com/serverpod/serverpod/pull/2816